pandas/tests/groupby/test_value_counts.py

```diff
@@ -79,3 +79,25 @@ def rebuild_index(df):
     # have to sort on index because of unstable sort on values
     left, right = map(rebuild_index, (left, right))  # xref GH9212
     tm.assert_series_equal(left.sort_index(), right.sort_index())
+
+
+@pytest.mark.parametrize("freq", ["1D", "2D", "1W", "1Y"])
+@pytest.mark.parametrize("size", [100, 1000])
+@pytest.mark.parametrize("frac", [0.1, 0.5, 1])
+def test_series_groupby_value_counts_with_grouper(freq, size, frac):
+    # GH28479
+    df = DataFrame.from_dict(
+        {
+            "date": date_range("2019-09-25", periods=size),
+            "name": np.random.choice(list("abcd"), size),
+        }
+    ).sample(frac=frac)
```

Comment:
Can you simplify this further? Using random data makes things more difficult than they need to be when reconstructing across test runs. If you can build with a very clear, reproducible test case first it would make things much easier

Reply:
> Can you simplify this further? Using random data makes things more difficult than they need to be when reconstructing across test runs. If you can build with a very clear, reproducible test case first it would make things much easier

I replace it with a variable-length slice to make it reproducible.